From 5ebfb24497bc03925dfaaab2014b4a8fd24c875f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 22 Jun 2021 20:38:58 +0300
Subject: [PATCH] [SW-46624] module_lkd: fix comms state mach abuse

Do not allow abusing the COMMS state machine by manipulating the
commands order.
Clear the execution result each time a PREP_DESC command is sent.

Change-Id: I9fe7fb2ffa1ee642e8c03ca069c8fdd588c5e333
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 3933183..decaf59 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -264,6 +264,7 @@
 		break;
 	case COMMS_PREP_DESC:
 	case COMMS_PREP_DESC_ELBI:
+		sts.exec_res = NOT_FOUND;
 		sts.next_size = sts.cmd.size;
 		/* no break */
 	case COMMS_CLR_STS:
